errors.go

```diff
@@ -13,68 +13,68 @@ import (
-	errServerNoMatchingSRTPProfile      = &FatalError{errors.New("client requested SRTP but we have no matching profiles")}
-	errServerRequiredButNoClientEMS     = &FatalError{errors.New("server requires the Extended Master Secret extension, but the client does not support it")}
-	errVerifyDataMismatch               = &FatalError{errors.New("expected and actual verify data does not match")}
-
-	errHandshakeMessageUnset             = &InternalError{errors.New("handshake message unset, unable to marshal")}
-	errInvalidFlight                     = &InternalError{errors.New("invalid flight number")}
-	errKeySignatureGenerateUnimplemented = &InternalError{errors.New("unable to generate key signature, unimplemented")}
-	errKeySignatureVerifyUnimplemented   = &InternalError{errors.New("unable to verify key signature, unimplemented")}
-	errLengthMismatch                    = &InternalError{errors.New("data length and declared length do not match")}
-	errNotEnoughRoomForNonce             = &InternalError{errors.New("buffer not long enough to contain nonce")}
-	errNotImplemented                    = &InternalError{errors.New("feature has not been implemented yet")}
-	errSequenceNumberOverflow            = &InternalError{errors.New("sequence number overflow")}
-	errUnableToMarshalFragmented         = &InternalError{errors.New("unable to marshal fragmented handshakes")}
+	errBufferTooSmall               = &TemporaryError{errors.New("buffer is too small")}                                        //nolint:goerr113
+	errContextUnsupported           = &TemporaryError{errors.New("context is not supported for ExportKeyingMaterial")}          //nolint:goerr113
+	errDTLSPacketInvalidLength      = &TemporaryError{errors.New("packet is too short")}                                        //nolint:goerr113
+	errHandshakeInProgress          = &TemporaryError{errors.New("handshake is in progress")}                                   //nolint:goerr113
+	errInvalidContentType           = &TemporaryError{errors.New("invalid content type")}                                       //nolint:goerr113
+	errInvalidMAC                   = &TemporaryError{errors.New("invalid mac")}                                                //nolint:goerr113
+	errInvalidPacketLength          = &TemporaryError{errors.New("packet length and declared length do not match")}             //nolint:goerr113
+	errReservedExportKeyingMaterial = &TemporaryError{errors.New("ExportKeyingMaterial can not be used with a reserved label")} //nolint:goerr113
+
+	errCertificateVerifyNoCertificate   = &FatalError{errors.New("client sent certificate verify but we have no certificate to verify")}                      //nolint:goerr113
+	errCipherSuiteNoIntersection        = &FatalError{errors.New("client+server do not support any shared cipher suites")}                                    //nolint:goerr113
```

Comment:
Should we wrap each error with a static error based on TemporaryError, FatalError, etc?

e.g
```go
var errTemporaryError = errors.New("Temporary Error")

var errBufferTooSmall = &TemporaryError{fmt.Errorf("%w: buffer is too small", errTemporaryError)} 
```

Reply:
It should be wrapped to implement `net.Error`.
`Temporary()` is required to handle errors in DoS-vulnerability-free way. (see https://github.com/pion/dtls/releases/tag/v2.0.0 `Fix unreachable error handling`)

Reply:
@Sean-Der I think `err113` lint should not be enabled.

https://github.com/Djarvur/go-err113#details
> error should not be created dynamically from scratch but by the wrapping the static (package-level) error.

I believe this is basically wrong from the Go's language design.
Using `fmt.Errorf("%w")` is just a simplified way of implementing Go 1.13 error interface. Having package specific error type and embed context of the error as a machine readable values must be allowed by default.

Reply:
@at-wat What do you think of something like?

```
      // TemporaryError indicates that the DTLS connection is still available, but the request was failed temporary.
      TemporaryError    = errors.New("A error that is temporary. Please retry")                
      errBufferTooSmall = fmt.Errorf("%w: buffer is too small", TemporaryError)         
```
The users can do `errors.Is(err, dtls.TemporaryError)`

With `goerr113` I found a lot of things we did wrong. I don't think we will see immediate gain, but long term I think it will really help consistency wise.

Reply:
> The users can do `errors.Is(err, dtls.TemporaryError)`

@Sean-Der It's not `net.Error` compatible. User should check both `err.(net.Error).Temporary()` (from underlying `net.Conn`) and `errors.Is(err, dtls.TemporaryError)`. If there is another layers of `net.Conn` user should check all error type of each package, otherwise it injects DoS vulnerability.

Reply:
Nice catch, sorry missed that nuance.

So personally my vote is keeping. `goerr113`. It is very valuable for packages that get less attention `srtp`, `datachannel` and `turn` being the big ones. I think for `dtls` a lot more effort has been put in, and it is fine to ignore them on a case by case basis.

Personally I am in favor of merging this PR. I respect your decision if you disagree, we can find a better solution!

Reply:
This PR looks good to me with these nolint comments. Linter selection thing is better to be discussed at https://github.com/pion/.goassets.